src/node_crypto.cc

```diff
@@ -3797,9 +3797,7 @@ void Hmac::HmacDigest(const FunctionCallbackInfo<Value>& args) {
 
   enum encoding encoding = BUFFER;
   if (args.Length() >= 1) {
-    encoding = ParseEncoding(env->isolate(),
-                             args[0]->ToString(env->isolate()),
-                             BUFFER);
+    encoding = ParseEncoding(env->isolate(), args[0], BUFFER);
```

Comment:
If I remember correctly, most functions just ignore invalid encoding values and use the default value instead. Currently, we are forcing the conversion to a string, but just because it is a string doesn't make it a valid encoding value. Adding a `CHECK` shouldn't do any harm, until we decide to finally drop support for `toString()` for consistency with other APIs. However, if we do not add a chack and the parameter is not a string (which won't happen now, but might as soon as we decide to drop support for `toString()`), there won't be any problems, the value will be ignored (`ParseEncoding` will return instantly).

The above only applies to `hmac` and `hash`. For `sign` and `verify`, the encoding parameter seems to always be `null`, so the `CHECK` would fail, right?